File: app/src/main/java/ua/syt0r/furiganit/ui/history/HistoryAdapter.java
package ua.syt0r.furiganit.ui.history;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

import ua.syt0r.furiganit.R;
import ua.syt0r.furiganit.model.entity.HistoryItem;

public class HistoryAdapter extends RecyclerView.Adapter<HistoryAdapter.HistoryViewHolder> {

    private List<HistoryItem> data = new ArrayList<>();

    @NonNull
    @Override
    public HistoryViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new HistoryViewHolder(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.adapter_history_item, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull HistoryViewHolder holder, int position) {
        HistoryItem historyItem = data.get(position);
        holder.textView.setText("");
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    void updateData(List<HistoryItem> history) {
        data = history;
        notifyDataSetChanged();
    }

    class HistoryViewHolder extends RecyclerView.ViewHolder {

        TextView textView;
        TextView dateTextView;

        HistoryViewHolder(@NonNull View itemView) {
            super(itemView);
            textView = itemView.findViewById(R.id.text);
            dateTextView = itemView.findViewById(R.id.date);
        }

    }

}

File: app/src/main/java/ua/syt0r/furiganit/app/history/HistoryAdapter.kt
package ua.syt0r.furiganit.app.history

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import java.util.ArrayList
import ua.syt0r.furiganit.R
import ua.syt0r.furiganit.model.entity.HistoryItem

class HistoryAdapter : RecyclerView.Adapter<HistoryAdapter.HistoryViewHolder>() {

    private var data: List<HistoryItem> = ArrayList()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HistoryViewHolder {
        return HistoryViewHolder(LayoutInflater.from(parent.context)
                .inflate(R.layout.adapter_history_item, parent, false))
    }

    override fun onBindViewHolder(holder: HistoryViewHolder, position: Int) {
        val historyItem = data[position]
        holder.textView.text = ""
    }

    override fun getItemCount(): Int {
        return data.size
    }

    fun updateData(history: List<HistoryItem>) {
        data = history
        notifyDataSetChanged()
    }

    inner class HistoryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var textView: TextView = itemView.findViewById(R.id.text)
        var dateTextView: TextView = itemView.findViewById(R.id.date)
    }

}
